kv/kvserver: TestStoreLoadReplicaQuiescent failed

kv/kvserver.TestStoreLoadReplicaQuiescent [failed](https://mesolite.cluster.engflow.com/invocations/default/46848b2e-d050-456a-904c-ea5cc978f312?testReportRun=3&testReportShard=40&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on release-25.2 @ [616957cfd997de48743591e06ede4b936237244c](https://github.com/cockroachdb/cockroach/commits/616957cfd997de48743591e06ede4b936237244c):


```
=== RUN   TestStoreLoadReplicaQuiescent
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestStoreLoadReplicaQuiescent3403640839
    test_log_scope.go:76: use -show-logs to present logs inline
    client_store_test.go:213: -- test log scope end --
test logs left over in: outputs.zip/logTestStoreLoadReplicaQuiescent3403640839
--- FAIL: TestStoreLoadReplicaQuiescent (8.03s)
=== RUN   TestStoreLoadReplicaQuiescent/lease-type=LeaseLeader
    client_store_test.go:182: 
        	Error Trace:	pkg/kv/kvserver/client_store_test.go:182
        	            				pkg/testutils/subtest.go:23
        	Error:      	Not equal: 
        	            	expected: 3
        	            	actual  : 1
        	Test:       	TestStoreLoadReplicaQuiescent/lease-type=LeaseLeader
--- FAIL: TestStoreLoadReplicaQuiescent/lease-type=LeaseLeader (1.49s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=3</code>
 - <code>shard=40</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStoreLoadReplicaQuiescent.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-62501

--- a/pkg/kv/kvserver/client_store_test.go
+++ b/pkg/kv/kvserver/client_store_test.go
@@ -171,15 +171,26 @@ func TestStoreLoadReplicaQuiescent(t *testing.T) {
 			// and the lease acquisition will trigger an election. Expire that lease
 			// and send a request that'll force a re-acquisition. This time, we should
 			// get a leader lease.
+			//
+			// However, the clock increment also expires store liveness support,
+			// which can cause the Raft leader to step down. If this happens, we may
+			// end up acquiring yet another expiration based lease -- but that's fine,
+			// as eventually, we'll upgrade to a leader lease; the test needs to wait this
+			// out.
 			manualClock.Increment(tc.Server(0).RaftConfig().RangeLeaseDuration.Nanoseconds())
 			incArgs := incrementArgs(key, int64(5))
 
 			testutils.SucceedsSoon(t, func() error {
-				_, err := kv.SendWrapped(ctx, tc.GetFirstStoreFromServer(t, 0).TestSender(), incArgs)
-				return err.GoError()
+				_, pErr := kv.SendWrapped(ctx, tc.GetFirstStoreFromServer(t, 0).TestSender(), incArgs)
+				if pErr != nil {
+					return pErr.GoError()
+				}
+				lease, _ = repl.GetLease()
+				if lease.Type() != roachpb.LeaseLeader {
+					return errors.Errorf("expected leader lease, got %s", lease.Type())
+				}
+				return nil
 			})
-			lease, _ = repl.GetLease()
-			require.Equal(t, roachpb.LeaseLeader, lease.Type())
 		}
 
 		switch leaseType {

